model/Versioned.php

```diff
@@ -1216,7 +1216,7 @@ public static function can_choose_site_stage($request) {
 	 */
 	public static function choose_site_stage(SS_HTTPRequest $request = null) {
 		if (!$request) {
-			throw new InvalidArgumentException("Request not found");
+			$request = Controller::curr()->getRequest();
```

Comment:
We should check `Controller::has_curr()` first since there isn't always a current controller.